### PR TITLE
 feat(LAMBDA-10166):Rule 2 - file size

### PR DESCRIPTION
**Pull request by melrakaibyxo**
## Jira LAMBDA Ticket ID:

[LAMBDA-10166](https://rapid-engineering.atlassian.net/jira/servicedesk/projects/LAMBDA/queues/issue/LAMBDA-10166)

## Description

Implementing Rule #2: "Single Responsibility: File Size Limit files to 400 lines of code"

## Demo Video

[Demo](https://www.loom.com/share/47ac0b5281614095a66ce6ee7acdbcf1)

## Checklist

- [x] I confirm that [Rapid Prototype IQB](https://docs.google.com/document/d/1TvF_Ll2BKEBvglHJW4OiYlM9PA6AtZy-eXhQagHB_UQ/edit#heading=h.vwb2v3h452be) has been checked before creating PR

### CODE REVIEW

**Comment by audalykh** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

Please consider using [Analyzer UI](https://confluence.devfactory.com/display/CodeReview/Analyzer+UI) extension to catch issues in pull requests.

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: lib/index.js [hunk 1]
```diff
@@ -14,11 +14,16 @@ module.exports = {
             ],
             plugins: ['rp', '@angular-eslint'],
             rules: {
-                "rp/rule-of-one": ["error", 1]
+              "rp/rule-of-one": ["error", 1],
```
Comment (audalykh) [inline lib/index.js:17]: <details>
<summary>:warning: <strong>This code is using `"` double-quotes to declare strings.</strong> (click for details)</summary>

A possible improvement would be to use `'` single-quote to declare all strings.  
  
This change would make the code consistent and easier to maintain.

**[More details](https://confluence.devfactory.com/x/8BhnFQ)**

## The improvement suggestion in this case
```js  
'rp/rule-of-one': ['error', 1],
```  
> NOTE: Consider applying the structure of this solution in other parts of the code, if there is another occurrence(s).
</details>

[//]: # (<a id="d88720c1-ecb2-4d57-986a-a7954c632b74" data-finding-id="359078128" href="none" codereview-uuid-reserved-tag data-business-id="d4071471-d3d4-4067-a020-eb1c00279a5a" data-analyzer="js" />)

File: lib/rules/single-responsibility-file-size.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+     meta: {
+         docs: {
+             description: 'Limit files to 400 lines of code',
+             url: 'https://angular.io/guide/styleguide#rule-of-one'
+         },
+         messages: {
+            exceed: 'QB Rule #2: File has too many lines ({{actual}}). Maximum allowed is {{max}}'
+         },
+     }
+ }
```
Comment (audalykh) [inline lib/rules/single-responsibility-file-size.js:22]: <details>
<summary>:warning: <strong>This code contains a statement without a semicolon.</strong> (click for details)</summary>

A possible improvement would be to keep the code consistency and insert the semicolon at the end of the line.  
  
This change would ensure the code follows the current code standard, making it easier for new developers to start without the need to learn a different convention and, thus, improving maintainability and keeping the code consistency.

**[More details](https://confluence.devfactory.com/x/lAsaF)**

## The improvement suggestion in this case
Consider adding the semicolon at the end of the line.  
  
```js  
};
```
</details>

[//]: # (<a id="6fec1eae-a399-4fad-83fb-c63f9018ee75" data-finding-id="337251220" href="none" codereview-uuid-reserved-tag data-business-id="d4071471-d3d4-4067-a020-eb1c00279a5a" data-analyzer="js" />)

**Review by audalykh — COMMENTED** [review]
Hi, I am the JavaScript expert who performed the review of this Pull Request. We have well-defined [coding and review standards](https://confluence.devfactory.com/display/CodeReview/Findings), and you will find below suggestions to improve the code quality, based on these guidelines.
My additional comments are in the Notes section. I hope you find them useful.

<table>
  <tr>
    <td>:no_entry_sign:<b>Severes:</b>  0</td>
    <td>:exclamation:<b>Reds:</b>  0</td>
    <td>:warning:<b>Yellows:</b>  2</td>
    <td>:white_check_mark:<b>Greens:</b>  0</td>
  </tr>
</table>

<details><summary>More information</summary>
  <table>
    <tr>
      <td>:no_entry_sign:</td>
      <td><b>Severe Findings</b> point out serious violations which can affect stability, performance, maintainability and therefore should be fixed.
        <ul>
          <li>Please note that the merge might have been blocked until all Severe Findings are fixed.</li>
          <li>While it is highly recommended to implement all findings only Severe Findings might have been blocking the merge</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>:exclamation:</td>
      <td><b>Red Findings</b> represent potential bugs and it is highly recommended to implement them.<br/>
      Not required for the approval of the PR
      </td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td><b>Yellows</b> are suggestions to improve the code, warnings and tips<br/>
      Not required for the approval of the PR, and should not be used for FTAR calculation.
      </td>
    </tr>
    <tr>
      <td>:white_check_mark:</td>
      <td><b>Green Findings</b> highlight good ideas and solutions</td>
    </tr>
    <tr>
      <td>:repeat:</td>
      <td><b>Repeated</b> instances of the same findings are counted just once and flagged with the repeat icon</td>
    </tr>
  </table>

  Feel free to reply to this comment, (make sure you `@tag` us) so we can start a conversation. The goal here is to learn and grow together and make our products better.

  > NOTE: This review **does not** replace your team's review._
</details>
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR increased your FTAR from 91.67 % to 92.31 %, good job, and for more quality improvements you can read about yellow findings.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>92.31 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>13</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>12</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>1</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Comment by audalykh** [comment]
You are not receiving code review NPS requests in your digest because you don't have your work email address specified in your [GitHub profile](https://github.com/melrakaibyxo).
<details>
<summary>More information</summary>
<table><tr><td>
To avoid this please follow the following steps:

1. Add your corporate email address on GitHub via [this link](https://github.com/settings/emails (a verification of an email is required))
2. Uncheck the "Keep my email address private" option
3. Switch to [profile settings](https://github.com/settings/profile) and select your corporate email address in the "Public email" drop-box
4. Click "Update profile"
</td></tr></table>
</details>
<br>

Hi, I'm Alexey Udalykh from [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) and I have just reviewed your PR.
<b>How likely are you to recommend the Code Review service to another developer?</b>

Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/rapid-qc/pull/1&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/rapid-qc/pull/1&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/rapid-qc/pull/1&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/rapid-qc/pull/1&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/rapid-qc/pull/1&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/rapid-qc/pull/1&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/rapid-qc/pull/1&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/rapid-qc/pull/1&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/rapid-qc/pull/1&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/rapid-qc/pull/1&entry.1163746017=10) \] Very likely

> TIP: Press `Ctrl` (or `⌘` in a Mac) to open the survey in a new tab.

**Comment by emil-cazacu-aurea** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

Please consider using [Analyzer UI](https://confluence.devfactory.com/display/CodeReview/Analyzer+UI) extension to catch issues in pull requests.

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

**Comment by emil-cazacu-aurea** [comment]
:white_check_mark: No new issues have been introduced since the previous review.

**Good work!**
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR increased your FTAR from 91.67 % to 92.31 %, good job, and for more quality improvements you can read about yellow findings.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>92.31 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>13</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>12</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>1</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Review by msretin — APPROVED** [review]

**Pull request merged**